docker-compose.yml

```diff
@@ -0,0 +1,211 @@
+version: '3.8'
+
+services:
+  angular-build:
+    build:
+      context: .
+      dockerfile: ./docker/Dockerfile
+    container_name: oppia-angular-build
+    volumes:
+      - .:/app/oppia
+      - shared_node_modules:/app/oppia/node_modules
+      - shared_third_party:/app/oppia/third_party
+      - shared_proto_files:/app/oppia/proto_files
+      - shared_js_ts_proto_files:/app/oppia/extensions/classifiers/proto
+      - shared_assets_dir:/app/oppia/assets
+      - shared_angular_build:/app/oppia/dist
+    healthcheck:
+      test: test "$(ls -A /app/oppia/dist)" || exit 1
+      timeout: 30s
+      interval: 30s
+      retries: 5
+      start_period: 20m
+    # added max-old-space-size to avoid out of memory error, check this link for more info
+    # https://stackoverflow.com/questions/62663167/dockerizing-react-in-production-mode-fatal-error-ineffective-mark-compacts-nea
+    command: >
+      sh -c '
+        trap "rm /app/oppia/node_modules/@angular/compiler-cli/ngcc/__ngcc_lock_file__" EXIT
+        export NODE_OPTIONS=--max_old_space_size=4096 && /app/oppia/node_modules/.bin/ng build --watch
+      '
+
+  webpack-compiler:
+    build:
+      context: .
+      dockerfile: ./docker/Dockerfile
```

Comment:
Do we need to mention which stage to use for this build i.e, which FROM statement to use?

Reply:
we cannot use targetted stage within the docker compose service, as this service is also dependent on proto files and other generated files, which are downloaded within the backend stage, so that's why we need to have whole Dockerfile, not targeting stage of dockerfile

Reply:
How would we know which container is it using?

Reply:
the frontend services do hava the proto files imports

Reply:
This doesn't answers the two questions I've asked:
> How would we know which container is it using? This is weird, so what's the use of a multi-stage building, isn't it making it more complicated now?

Reply:
corrected: added the target build stage in the services! thank you